qa/SidechainTestFramework/account/ac_use_smart_contract.py

```diff
@@ -75,43 +78,62 @@ def __init__(self, contract_path: str):
                     tx_hash: The transaction hash of the transaction
+                :param args: the arguments to call the function with, in the correct order
+                :param node: The side chain node to use for rpc/http
+                :param functionName: The name of the function to call. Has to include parentheses and params without spaces (e.g. `someFunction(uint256,string)`
+                :param call_method: The type of transaction to create
+                :param fromAddress: the sender address to use
+                :param toAddress: the address of the smart contract instance to use
+                :param gasLimit: The gasLimit to use
+                :param gasPrice: The gasPrice to use
+                :param nonce: The nonce to use. If not given, the nonce will be retrieved via rpc
+                :param tag: The block tag to use when calling rpc methods
+                :param value: The value to use
+                :param maxFeePerGas: maxFeePerGas in EIP1559
+                :param maxPriorityFeePerGas: maxPriorityFeePerGas in eip1559
 
         """
-        j = {
-            "from": format_eoa(fromAddress),
-            "to": format_eoa(toAddress),
-            "nonce": self.__ensure_nonce(node, fromAddress, nonce, tag),
-            "gasLimit": gasLimit,
-            "gasPrice": gasPrice,
-            "value": value,
-            "data": self.raw_encode_call(functionName, *args)
-        }
-        request = json.dumps(j)
-        response = node.transaction_createLegacyTransaction(request)
-        return response["result"]["transactionId"]
+        data = self.raw_encode_call(functionName, *args)
+        nonce = self.__ensure_nonce(node, fromAddress, nonce, tag)
+        chain_id = self.__ensure_chain_id(node)
+        if call_method is CallMethod.RPC_LEGACY:
+            response = node.rpc_eth_signTransaction(
+                self.__make_legacy_sign_payload(from_addr=fromAddress, to=toAddress, nonce=nonce, gas_price=gasPrice,
+                                                gas=gasLimit, data=data, value=value))
+            print(response)
+            response = node.rpc_eth_sendRawTransaction(response['result'])
+            print(response)
+            return response['result']
+        elif call_method is CallMethod.RPC_EIP155:
+            response = node.rpc_eth_signTransaction(
+                self.__make_eip155_sign_payload(from_addr=fromAddress, to=toAddress, chain_id=chain_id, nonce=nonce,
+                                                gas_price=gasPrice, gas=gasLimit, data=data, value=value))
+            print(response)
+            response = node.rpc_eth_sendRawTransaction(response['result'])
+            print(response)
+            return response["result"]
+        elif call_method is CallMethod.RPC_EIP1559:
+            response = node.rpc_eth_signTransaction(
+                self.__make_eip1559_sign_payload(from_addr=fromAddress, to=toAddress, chain_id=chain_id, nonce=nonce,
```

Comment:
Some code deduplication would be nice, all the if branches are almost identical.